util/chunk/column.go

```diff
@@ -242,33 +242,56 @@ const (
 	sizeGoDuration = int(unsafe.Sizeof(time.Duration(0)))
 )
 
-// preAlloc allocates space for a fixed-length-type slice and resets all slots to null.
-func (c *Column) preAlloc(length, typeSize int) {
-	nData := length * typeSize
-	if len(c.data) >= nData {
-		c.data = c.data[:nData]
+// resize resizes the column so that it contains n elements, only valid for fixed-length types.
+func (c *Column) resize(n, typeSize int) {
+	sizeData := n * typeSize
+	if cap(c.data) >= sizeData {
+		(*reflect.SliceHeader)(unsafe.Pointer(&c.data)).Len = sizeData
 	} else {
-		c.data = make([]byte, nData)
+		c.data = make([]byte, sizeData)
 	}
 
-	nBitmap := (length + 7) >> 3
-	if len(c.nullBitmap) >= nBitmap {
-		c.nullBitmap = c.nullBitmap[:nBitmap]
-		for i := range c.nullBitmap {
-			// resets all slots to null.
-			c.nullBitmap[i] = 0
-		}
+	sizeNulls := (n + 7) >> 3
+	if cap(c.nullBitmap) >= sizeNulls {
+		(*reflect.SliceHeader)(unsafe.Pointer(&c.nullBitmap)).Len = sizeNulls
 	} else {
-		c.nullBitmap = make([]byte, nBitmap)
+		c.nullBitmap = make([]byte, sizeNulls)
 	}
 
-	if c.elemBuf != nil && len(c.elemBuf) >= typeSize {
-		c.elemBuf = c.elemBuf[:typeSize]
+	if cap(c.elemBuf) >= typeSize {
+		(*reflect.SliceHeader)(unsafe.Pointer(&c.elemBuf)).Len = typeSize
 	} else {
 		c.elemBuf = make([]byte, typeSize)
 	}
 
-	c.length = length
+	c.length = n
+}
+
+// reserve makes the column capacity be at least enough to contain n elements.
+func (c *Column) reserve(n, estElemSize int) {
```

Comment:
1. Add a comment for `estElemSize `
2. Add a comment `only valid for var-length types`?

Reply:
fixed.